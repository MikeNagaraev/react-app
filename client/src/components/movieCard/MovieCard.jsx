import React from "react";

export default class MovieCard extends React.Component {
    render() {
        return (
            <div className="movieCard">
                <div className="movieCard-img">
                    <img src={this.props.picture} alt=""/>
                </div>
                <div className="movieCard-description">
                    <div className="movieCard-description-line">
                        <span className="movieCard-title">{this.props.title}</span>
                        <span className="movieCard-year">{this.props.year}</span>
                    </div>
                    <div className="movieCard-description-line">
                        <span className="movieCard-genre">{this.props.genre}</span>
                    </div>
                </div>
            </div>
        )
    }
}