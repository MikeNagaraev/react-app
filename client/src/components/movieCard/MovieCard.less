.movieCard {
    background: @grey2;
    width: 25rem;
    margin: 4rem;
    min-height: 35rem;
    &-img {
        width: 100%;
        height: 30rem;
        img {
            width: 100%;
            height: 100%;
        }
    }
    &-description {
        padding: 1rem;
        &-line {
            width: 100%;
            clear: both;
        }
    }
    &-title {
        text-transform: uppercase;
        color: @black1;
        font-weight: bold;
        float: left;
    }
    &-year {
        float: right;
        border: 1px solid @grey2;
        padding: 0.3rem;
    }
    &-genre {
        color: @grey1;
    }
}

@media only screen and (max-width: 660px) {
    .movieCard {
        width: 51%;
        margin: 2rem auto;
    }
}