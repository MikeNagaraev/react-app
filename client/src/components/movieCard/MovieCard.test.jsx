import * as React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import MovieCard from './MovieCard';

describe('MovieCard component', () => {
    beforeAll(() => {
        configure({ adapter: new Adapter() });
    });

    it('should render component', () => {
        const enzymeWrapper = mount(<MovieCard />);
        expect(enzymeWrapper.find('.movieCard')).toHaveLength(1);
    });
    
    it('renders correctly snapshot', () => {
        const tree = renderer.create(
            <MovieCard title="Film 1" year="2004" genre="Action" picture="/assets/images/pulp.jpg" />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
