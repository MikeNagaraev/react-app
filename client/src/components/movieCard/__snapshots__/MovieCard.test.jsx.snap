// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component renders correctly snapshot 1`] = `
<div
  className="movieCard"
>
  <div
    className="movieCard-img"
  >
    <img
      alt=""
      src="/assets/images/pulp.jpg"
    />
  </div>
  <div
    className="movieCard-description"
  >
    <div
      className="movieCard-description-line"
    >
      <span
        className="movieCard-title"
      >
        Film 1
      </span>
      <span
        className="movieCard-year"
      >
        2004
      </span>
    </div>
    <div
      className="movieCard-description-line"
    >
      <span
        className="movieCard-genre"
      >
        Action
      </span>
    </div>
  </div>
</div>
`;
